src/utils/app.utils.ts

```diff
@@ -0,0 +1,68 @@
+import { EventName, EventStatus, StatusMessages } from 'src/common/constants';
+import { FormattedData } from 'src/common/types/webhook.event.types';
+
+export function isPayloadValid(payload: any): boolean {
+  return (
+    payload?.action &&
+    payload?.pull_request &&
+    payload?.repository &&
+    payload?.sender &&
+    payload?.installation
+  );
+}
+
+export function createFormattedData(
+  partialData: Partial<FormattedData>,
+): FormattedData {
+  const defaultData: FormattedData = {
+    token: '',
+    eventId: '',
+    backendOptions: { organizationPractices: null },
+    context: {
+      eventName: EventName.PULL_REQUEST,
+      payload: {},
+      issue: { owner: '', repo: '', number: 1 },
+      repo: { owner: '', repo: '' },
+      sha: '',
+      ref: '',
+      workflow: '',
+      action: '',
+      actor: '',
+      job: '',
+      runNumber: 1,
+      runId: 1,
+      apiUrl: '',
+      serverUrl: '',
+      graphqlUrl: '',
+    },
+  };
+
+  return {
+    ...defaultData,
+    ...partialData,
+    context: {
+      ...defaultData.context,
+      ...partialData.context,
+    },
+  };
```

Comment:
PrimathonAICodeReviewer

The `createFormattedData` function seems to have a complex structure. It might be beneficial to refactor this function into smaller, more focused functions to improve readability and maintainability.



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->